### PR TITLE
feat: make reserve capacity configurable and dynamic

### PR DESCRIPTION
**Pull request by martinconic**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Implementation of points 1 and 2 from [#4842](https://github.com/ethersphere/bee/issues/4842) for [SWIP-21](https://github.com/ethersphere/SWIPs/pull/56/files)

### CODE REVIEW

**Review by istae — CHANGES_REQUESTED** [review]

File: pkg/storer/internal/reserve/reserve.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ type Reserve struct {
 func New(
 	baseAddr swarm.Address,
 	st transaction.Storage,
+	maxCapacityHeight int,
```
Comment (istae) [inline pkg/storer/internal/reserve/reserve.go:48]: internal reserve does not need to know about maxCapacityHeight

File: pkg/storer/reserve.go [hunk 2]
```diff
@@ -28,8 +28,13 @@ const (
 	reserveUnreserved   = "reserveUnreserved"
 	batchExpiry         = "batchExpiry"
 	batchExpiryDone     = "batchExpiryDone"
+
+	MaxReserveCapacityHeight = 1
```
Comment (istae) [inline pkg/storer/reserve.go:32]: reserve does not need to know about maxHeight

File: pkg/storer/internal/reserve/reserve.go (hunk 1 above)
Comment (martinconic) [reply pkg/storer/internal/reserve/reserve.go]: fixed

File: pkg/storer/reserve.go (hunk 2 above)
Comment (martinconic) [reply pkg/storer/reserve.go]: fixed

File: cmd/bee/cmd/start.go [hunk 3]
```diff
@@ -335,6 +335,7 @@ func buildBeeNode(ctx context.Context, c *command, cmd *cobra.Command, logger lo
 		WhitelistedWithdrawalAddress:  c.config.GetStringSlice(optionNameWhitelistedWithdrawalAddress),
 		TrxDebugMode:                  c.config.GetBool(optionNameTransactionDebugMode),
 		MinimumStorageRadius:          c.config.GetUint(optionMinimumStorageRadius),
+		ReserveCapacity:               1 << (22 + c.config.GetInt(optionReserveCapacityHeight)),
```
[the remark this answers is not in the document]
Comment (martinconic) [reply cmd/bee/cmd/start.go]: changed and simplified

File: pkg/node/node.go [hunk 4]
```diff
@@ -354,13 +353,21 @@ func NewBee(
 	var batchStore postage.Storer = new(postage.NoOpBatchStore)
 	var evictFn func([]byte) error
 
+	var reserveCapacity int
+
+	if o.ReserveCapacityDoubling > 0 && o.ReserveCapacityDoubling <= 1 {
```
Comment (gacevicljubisa) [inline pkg/node/node.go:358]: Probably it should be `>=` in `o.ReserveCapacityDoubling > 0`. Default is 0.

**Review by gacevicljubisa — CHANGES_REQUESTED** [review]

**Review by istae — APPROVED** [review]

**Review by gacevicljubisa — APPROVED** [review]

**Pull request merged**